File: python/src/day01/main.py
def main():
    with open('./input.txt', 'r') as f: input = f.read().split('\n')
    print(f"Silver: {silver(input)}")
    print(f"Gold: {gold(input)}")

def silver(input):
    max = 0
    temp = 0
    for x in input:
        if x == '':
            if temp > max:
                max = temp
            temp = 0
        else:
            temp += int(x)
    return max

def gold(input):
    max = []
    temp = 0
    for x in input:
        if x == '':
            if len(max) < 3:
                max.append(temp)
                max = sorted(max)
            else:
                if max[0] < temp:
                    max[0] = temp
                    max = sorted(max)
            temp = 0
        else:
            temp += int(x)
    return sum(max)

main()

File: rust/day01/src/main.rs
fn main() {
    let input: &str = include_str!("../input.txt");
    println!("Silver: {}", silver(input));
    println!("Gold: {}", gold(input));
}

fn silver(input: &str) -> i32 {
    input
        .split("\n\n")
        .map(|x| x.lines().map(|y| y.parse::<i32>().unwrap()).sum())
        .max()
        .unwrap_or(0)
}

fn gold(input: &str) -> i32 {
    let mut totals: Vec<i32> = input
        .split("\n\n")
        .map(|x| x.lines().map(|y| y.parse::<i32>().unwrap()).sum())
        .collect();
    totals.sort();
    totals.iter().rev().take(3).sum()
}